pkg/capabilities/triggers/mercury_trigger.go

```diff
@@ -91,29 +100,35 @@ func (o *MercuryTriggerService) ProcessReport(reports []mercury.FeedReport) erro
 		if !ok {
 			return fmt.Errorf("no registration for %s", triggerID)
 		}
-		ch <- capabilityResponse
+
+		select {
+		case ch <- capabilityResponse:
+		default:
+			o.log.Warn("dropping response due to slow consumer", "triggerID", triggerID)
```

Comment:
Rather than dropping in this case, why not just return an error? That means the caller can choose how to handle it.